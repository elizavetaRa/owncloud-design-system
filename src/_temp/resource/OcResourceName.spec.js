import { shallowMount } from "@vue/test-utils"

import Name from "./OcResourceName.vue"

const fullPath = "images/nature/forest.jpg"

describe("OcResourceName", () => {
  it("displays full path", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath,
        isPathDisplayed: true,
      },
    })

    expect(wrapper.find(".oc-resource-path").text()).toMatch("â€¦/nature/")
    expect(wrapper.find(".oc-resource-basename").text()).toMatch("forest")
    expect(wrapper.find(".oc-resource-extension").text()).toMatch(".jpg")
    expect(wrapper.find(".oc-resource-name").attributes(["uk-tooltip"])).toMatch(
      "images/nature/forest.jpg"
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("displays only direct parent", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: "Documents/notes.txt",
        isPathDisplayed: true,
      },
    })

    expect(wrapper.find(".oc-resource-path").text()).toMatch("Documents/")
    expect(wrapper.find(".oc-resource-basename").text()).toMatch("notes")
    expect(wrapper.find(".oc-resource-extension").text()).toMatch(".txt")
    expect(wrapper).toMatchSnapshot()
  })

  it("hides the path if not enabled", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath,
      },
    })

    expect(wrapper.findAll(".oc-resource-path").length).toBe(0)
    expect(wrapper.find(".oc-resource-basename").text()).toMatch("forest")
    expect(wrapper.find(".oc-resource-extension").text()).toMatch(".jpg")
    expect(wrapper).toMatchSnapshot()
  })

  it("doesn't add extension to hidden folder", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: ".hidden",
      },
    })

    expect(wrapper.find(".oc-resource-basename").text()).toMatch(".hidden")
    expect(wrapper.findAll(".oc-resource-extension").length).toBe(0)
    expect(wrapper).toMatchSnapshot()
  })
})
