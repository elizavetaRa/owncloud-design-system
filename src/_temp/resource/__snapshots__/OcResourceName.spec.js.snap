// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcResourceName displays full path 1`] = `<div resource-path="images/nature/forest.jpg" resource-name="…/nature/forest.jpg" uk-tooltip="images/nature/forest.jpg" class="oc-resource-name oc-text-truncate"><span class="oc-resource-path">…/nature/</span><span class="oc-resource-basename">forest</span><span class="oc-resource-extension">.jpg</span></div>`;

exports[`OcResourceName displays only direct parent 1`] = `<div resource-path="Documents/notes.txt" resource-name="Documents/notes.txt" class="oc-resource-name oc-text-truncate"><span class="oc-resource-path">Documents/</span><span class="oc-resource-basename">notes</span><span class="oc-resource-extension">.txt</span></div>`;

exports[`OcResourceName doesn't add extension to hidden folder 1`] = `
<div resource-path=".hidden" resource-name=".hidden" class="oc-resource-name oc-text-truncate">
  <!----><span class="oc-resource-basename">.hidden</span>
  <!---->
</div>
`;

exports[`OcResourceName hides the path if not enabled 1`] = `
<div resource-path="images/nature/forest.jpg" resource-name="forest.jpg" class="oc-resource-name oc-text-truncate">
  <!----><span class="oc-resource-basename">forest</span><span class="oc-resource-extension">.jpg</span>
</div>
`;
